### ISSUE TITLE
Build issues with Xcode 8 beta 3

### ISSUE THREAD

**Issue by RetepV** [bignerdranch/CoreDataStack#155]
### Feature Request

I am trying to build with Xcode 8 beta 3, while keep using Swift 2.3, and I get a few build errors. Can you fix them, so that I can get the fixes through 'pod install'?

There are issues in CoreDataStack.swift and in CoreDataModelable.swift.

### Actual behavior

**Build errors in CoreDataStack.swift:**
```
    let _ = try? fm.removeItemAtURL(storeURL.URLByAppendingPathComponent("-shm"))
                     (!)Value of optional type 'NSURL?' not unwrapped; did you mean to use '!' or '?'?
    let _ = try? fm.removeItemAtURL(storeURL.URLByAppendingPathComponent("-wal"))
                     (!)Value of optional type 'NSURL?' not unwrapped; did you mean to use '!' or '?'?
```
```URLByAppendingPathComponent``` returns an optional which should be unwrapped or handled correctly. Quick (but possibly unsafe) fix:
```
    let _ = try? fm.removeItemAtURL(storeURL.URLByAppendingPathComponent("-shm")!)
    let _ = try? fm.removeItemAtURL(storeURL.URLByAppendingPathComponent("-wal")!)
```

**Build error in CoreDataModelable.swift:**
```
    var error: NSError? = nil
    let count = context.countForFetchRequest(fetchReqeust, error: &error)
                     (!)Extra argument 'error' in call
    if let error = error { throw error }
    guard count != NSNotFound else { return 0 }
    return count
```
That should probably look something like this:
```
    var count: Int = NSNotFound
    do {
        count = try context.countForFetchRequest(fetchReqeust)
    }
    catch let error {
        throw error
    }
    guard count != NSNotFound else { return 0 }
    return count
```

Thank you and cheers!
Peter

**Comment by RetepV** [comment]
Note that these are quick fixes. It made my app compile, and it seems to run, but I'm not even close to having completely tested it.

**Comment by rcedwards** [comment]
Thanks @RetepV we are in the process of creating a Swift 2.3 compatible version of CoreDataStack along with Swift 3.0 as well (#152)

I will keep this issue open to track Swift 2.3 compatibility.

**Comment by nick3389** [comment]
Is the swift 3 branch functional? I have a project in swift 3 but i cannot use BNR master branch.

**Comment by rcedwards** [comment]
@nick3389 no the Swift 3 branch is a work in progress.

Swift 3 will become the default on master upon its completion and a Swift 2.3 branch will also be available for those not updating to 3.0.
